lib/puppet/ssl/certificate_request.rb

```diff
@@ -267,7 +267,17 @@ def extension_request_attribute(options)
 
     if options[:dns_alt_names]
       names = options[:dns_alt_names].split(/\s*,\s*/).map(&:strip) + [name]
-      names = names.sort.uniq.map {|name| "DNS:#{name}" }.join(", ")
+
+      names = names.sort.map do |name|
+        if (!name.start_with? "IP:") && (!name.start_with? "DNS:")
```

Comment:
You don't need parens around each clause, though you should put them around each arg to `start_with?`.